src/read.rs

```diff
@@ -1257,6 +1256,146 @@ impl<R: Read + Seek> ZipArchive<R> {
     }
 }
 
+impl<R> ZipArchive<R>
+where
+    R: Read + Seek,
+{
+    /// Search for a file entry by name
+    pub fn by_name_new(
```

Comment:
Change `_new` to something more descriptive in these function names. Maybe `_generic`.

Reply:
Done!